src/coreclr/src/vm/jitinterface.cpp

```diff
@@ -8881,76 +8884,76 @@ CORINFO_METHOD_HANDLE CEEInfo::resolveVirtualMethodHelper(CORINFO_METHOD_HANDLE
 
     MethodDesc* pDevirtMD = nullptr;
 
-    TypeHandle DerivedClsHnd(derivedClass);
-    MethodTable* pDerivedMT = DerivedClsHnd.GetMethodTable();
-    _ASSERTE(pDerivedMT->IsRestored() && pDerivedMT->IsFullyLoaded());
+    TypeHandle ObjClassHnd(info->objClass);
+    MethodTable* pObjMT = ObjClassHnd.GetMethodTable();
+    _ASSERTE(pObjMT->IsRestored() && pObjMT->IsFullyLoaded());
 
     // Can't devirtualize from __Canon.
-    if (DerivedClsHnd == TypeHandle(g_pCanonMethodTableClass))
+    if (ObjClassHnd == TypeHandle(g_pCanonMethodTableClass))
     {
-        return nullptr;
+        return false;
     }
 
     if (pBaseMT->IsInterface())
     {
 
 #ifdef FEATURE_COMINTEROP
         // Don't try and devirtualize com interface calls.
-        if (pDerivedMT->IsComObjectType())
+        if (pObjMT->IsComObjectType())
         {
             return nullptr;
         }
 #endif // FEATURE_COMINTEROP
 
         // Interface call devirtualization.
         //
-        // We must ensure that pDerivedMT actually implements the
+        // We must ensure that pObjMT actually implements the
         // interface corresponding to pBaseMD.
-        if (!pDerivedMT->CanCastToInterface(pBaseMT))
+        if (!pObjMT->CanCastToInterface(pBaseMT))
```

Comment:
Perhaps this PR could be a better place to introduce/evaluate [this change](https://github.com/dotnet/runtime/pull/43668/files#r512813024)

Reply:
We can do that by amending your PR, or as a subsequent PR.

Hopefully what I've done here sets things up so that change is more straightforward.